### ISSUE TITLE
Bug - Ember animated container content won't rerender

### ISSUE THREAD

**Issue by jenweber** [cardstack/cardstack#1049]
## Problem:
When animated-if is used inside an AnimatedContainer, the contents don't rerender when values change. This blocks a lot of animation work.

> Assertion Failed: Could not find component named "-dynamic-element" (no component or template with that name was found)

## Escape hatch

Could change static components to false "fixes" the bug

Cons are that we don't see when there are missing files, could reveal other bugs.

## To do
- [x] need to try and see if we can reproduce in a regular Octane app. Reproduced here https://github.com/jenweber/embroider-percy-debug/tree/animated-debug
- [x] Turn embroider off and see if the minimal repro works. Works normally.
- [ ] turn off staticComponents in the cardhost app and see how far that gets us (this helps)
- [x] try classic Ember w/Embroider and see if it works (it does)
- [ ] Where does `-dynamic-element` come from? 
- [ ] The `element` helper is null in AnimatedContainer. Where does that come from? It comes from ember-element-helper dependency https://github.com/tildeio/ember-element-helper

## Debugging tips
search for staticComponentspackages/core/src/app.ts
controls whether imports get pushed onto the stack (appJSAsset)look at the assets file in the build

When static components is on:
resolver.ts, look at astTransformer
Can't easily look at build output bc the template compiling is done by webpack, so use debuggerlook at resolveMustache. Adds imports to templates. ~~Seems to be including the component because it's not missing, just broken~~ AnimatedContainer is broken on initial render

Things to look for
- animated if being used by itself- renders correctly on the first try, but not on changes- seems like something is weird about component imports
- check for multiple copies of ember-animated. Should be ok, but could check- use will's branch. the thing that is not working is responsibility of animated-if, used inside an AnimatedContainer. Contents of if don't rerender when values change!- 

ruled out:
- infinite animation- look at redneredChildren in animated-each. Doesn't seem to be running
- positional params seemed ok when Ed looked at it

**Comment by jenweber** [comment]
The issue was a problem with `ember-element-helper` that was fixed in the 0.2.0 release. That version bump has already been merged in ember-animated, so once a release is cut, I think we will be good to go.

See minimal repro at
https://github.com/ember-animation/ember-animated/issues/182

**Comment by jenweber** [comment]
Ongoing issue when pinned to `github:jenweber/ember-animated#1f812240be876d91361ad9434adbf0cc78178fff`

```
-private/bounds.js: unable to use this non-serializable babel config in this node process

        in /tmp/broccoli-9577BDiOWouUZUEA/out-01-funnel

        at broccoli-persistent-filter:Babel > [Babel: ember-animated] (Babel: ember-animated)
```

**Comment by ef4** [comment]
This is fixed via embroider 0.10.0.

**Comment by jenweber** [comment]
Closed by #1122
